src/evidently/collector/app.py

```diff
@@ -89,25 +87,25 @@ async def reference(
     return {}
 
 
-@post("/{id:str}/data")
+@post("/{id:str}/data", sync_to_thread=True)
 async def data(
     id: Annotated[str, Parameter(description="Collector ID")],
-    request: Request,
-    service: CollectorServiceConfig,
-    storage: CollectorStorage,
+    data: Any,
+    service: Annotated[CollectorServiceConfig, Dependency(skip_validation=True)],
+    storage: Annotated[CollectorStorage, Dependency(skip_validation=True)],
 ) -> Dict[str, str]:
     if id not in service.collectors:
         raise HTTPException(status_code=404, detail=f"Collector config with id '{id}' not found")
     async with storage.lock(id):
-        storage.append(id, await request.json())
+        storage.append(id, data)
```

Comment:
So far I could only see an in-memory implementation of this interface, but seeing as this isn't async, it could become problematic if a blocking implementation is added. This would probably be better as a fully async service as well